src/emit.iced

```diff
@@ -113,6 +144,98 @@ exports.GoEmitter = class GoEmitter
     @untab()
     @output "}"
 
+  deep_copy : ( {t, val}) ->
+    if typeof(t) is 'string'
+      @deep_copy_simple { t, val }
+    else if typeof(t) isnt 'object'
+      @output "ERROR"
+    else if Array.isArray(t)
+      if t[0]? then @output "ERROR"
+      else @deep_copy_pointer { t , val }
+    else if t.type is 'array'
+      @deep_copy_array { t, val }
+    else if t.type is 'map'
+      @deep_copy_map { t, val }
+    else
+      @output "ERROR"
+
+  deep_copy_simple : ({t, val}) ->
+    if t not in [ 'boolean', 'long', 'float', 'double', 'string', 'int' ] then val += ".DeepCopy()"
+    else if t is 'bytes' then val = "append([]byte{}, #{val}...)"
```

Comment:
It might be better to use `append([]byte(nil), #{val}...)`. Because the way you have it `len(val) == 0` then you end up with a `!=nil` but `len==0` list (devil's list).
meh : https://play.golang.org/p/0tfT3-89ck
good: https://play.golang.org/p/IDJ44u7QXK